source/input.cpp

```diff
@@ -704,9 +767,33 @@ extern "C" BOOL WINAPI HookClipCursor(const RECT *lpRect)
 {
 	s_last_clip_cursor = (lpRect != nullptr) ? *lpRect : RECT {};
 
+	// Some applications clip the mouse cursor, so disable that while we want full control over mouse input
 	if (is_blocking_mouse_input())
-		// Some applications clip the mouse cursor, so disable that while we want full control over mouse input
-		lpRect = nullptr;
+	{
+		// Application tried to set a restrictive clip rect, expand it to the entire window if applicable
+		if (lpRect != nullptr)
+		{
+			lpRect = nullptr;
+
+			// Most of the time when the ReShade UI is open, users will want the mouse confined to the window.
+			// 
+			//   nb: Add a config parameter for this?
+			//     (Previous to this, ReShade was letting the cursor move to other monitors)
```

Comment:
I do indeed use this behavior from time to time (e.g. having a code editor on my second monitor and moving back and forth between that and the game window on the main monitor), so would be in favor of leaving out this change.

Reply:
I have reverted this.